driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -357,15 +489,201 @@ public T get(Object... primaryKey) {
+            }
+
+            /**
+             * Get the TTL value configured in the object.
+             *
+             * @return the TTL value.
+             */
+            public int getValue() {
+                return this.ttlValue;
+            }
+
+            void appendTo(Insert.Options usings) {
+                usings.and(QueryBuilder.ttl(QueryBuilder.bindMarker()));
+            }
+
+            void appendTo(Delete.Options usings) {
+                usings.and(QueryBuilder.ttl(QueryBuilder.bindMarker()));
+            }
+
+            void bindInStatement(BoundStatement bs, int i) {
+                bs.setInt(i, this.ttlValue);
+            }
+
+            boolean isValidFor(QueryType qt){
+                return qt == QueryType.SAVE;
+            }
+
+        }
+
+        static class Timestamp extends Option {
+
+            private long tsValue;
+
+            Timestamp(long value) {
+                this.tsValue = value;
+            }
+
+            /**
+             * Get the TIMESTAMP value configured in the object.
+             *
+             * @return the TIMESTAMP value.
+             */
+            public long getValue() {
+                return this.tsValue;
+            }
+
+            void appendTo(Insert.Options usings){
+                usings.and(QueryBuilder.timestamp(QueryBuilder.bindMarker()));
+            }
+
+            void appendTo(Delete.Options usings) {
+                usings.and(QueryBuilder.timestamp(QueryBuilder.bindMarker()));
+            }
+
+
+            boolean isValidFor(QueryType qt){
+                return qt == QueryType.SAVE || qt == QueryType.DEL;
+            }
+
+            void bindInStatement(BoundStatement bs, int i){
+                bs.setLong(i, this.tsValue);
+            }
+
+        }
+
+        /**
+         * Creates a new SaveOptions object for adding a TTL value in a save
+         * or delete operation.
+         *
+         * @param value the value to use for the operation.
+         * @return the SaveOptions object configured to set a TTL value to a
+         * save or delete operation.
+         */
+        public static Option ttl(int value) {
+            return new Ttl(value);
+        }
+
+        /**
+         * Creates a new SaveOptions object for adding a TIMESTAMP value in a save
+         * or delete operation.
+         *
+         * @param value the value to use for the operation.
+         * @return the SaveOptions object configured to set a TIMESTAMP value to a
+         * save or delete operation.
+         */
+        public static Option timestamp(long value) {
+            return new Timestamp(value);
+        }
+
+        abstract void appendTo(Insert.Options usings);
+
+        abstract void appendTo(Delete.Options usings);
+
+        abstract void bindInStatement(BoundStatement bs, int i);
+
+        abstract boolean isValidFor(QueryType qt);
+
+        public int hashCode() {
+            return this.getClass().hashCode();
```

Comment:
Since hashCode is generated off of the Class, couldn't this be a problem if you had two separate queries with the same Options but different option values?  (I.e. one query with TTL of 300, and another of 600)